pagopa-archive/io-app-cie
# [#175150462] Added react native mini-app to test support for APDU protocol in pkg react-native-nfc-manager

**Short Description**
I've added a mini react native app to the bare io-template-typescript to test basic communication with an NFC enabled mobile phone via react-native-nfc-manager package. The app executes a standard test from the ICAO test suite in [Test Standard for eMRTD – Part 3](https://www.icao.int/Security/FAL/TRIP/Documents/TR%20-%20RF%20and%20Protocol%20Testing%20Part%203%20V2.11.pdf) for MRTD application in CIE.

**List of changes proposed in this pull request**
- Adoption of package react-native-nfc-manager in the IO-APP tech stack/dependencies for basic communication via APDU protocol with NFC enabled mobile phones
- Adoption of [Buffer](https://nodejs.org/api/buffer.html) library for the encoding-decoding-transcoding of binary APDU messages

**How To test**
1. Run the application on an NFC enabled mobile phone (Android or IOS 13+),
2. Click on RUN ICAO TEST CASE 7816_A_1
3. Move the CIE near the mobile phone
4. Check in the rendered text box that the result is 9000 (in hexadecimal encoding)

**Note**
I tested it only on Android. See how in the attached screenshots
![Run Application](https://user-images.githubusercontent.com/8901084/96428478-1665c380-1200-11eb-8f43-2c5272940379.jpg)
![Run the test and move the CIE near the mobile phone](https://user-images.githubusercontent.com/8901084/96428576-385f4600-1200-11eb-9b78-7f161b3a62c1.jpg)
![Check test results in the proper text box](https://user-images.githubusercontent.com/8901084/96428618-47de8f00-1200-11eb-9ebb-686c6cfb1772.jpg)
